torch/csrc/jit/codegen/cuda/runtime/tensor_core.cu

```diff
@@ -0,0 +1,410 @@
+
+// Arch guard, Volta + only
+#if (defined(__CUDA_ARCH__) && (__CUDA_ARCH__ >= 700))
+
+namespace mma_instruction {
+
+// MMA instruction wrappers (sm_70+):
+// The instruction wrappers below are quarter-warp macros, which currently
+// nvfuser
+//  doesn't explicitly model. So they are currently only meant to be
+//  used as building blocks in warp level mma macros
+
+//  8x8x4 mma instruction, per quarter warp (8 threads), fp16 accumulate
+//  per thread register:
+//   A[4] x B[4] -> C[8]
+__device__ inline void mmaM8n8k4(__half* C, __half* A, __half* B) {
```

Comment:
Let's not support fp16 accumulate at this point. It can be really dangerous in training.